src/common/viz/widgets/BrowserAssetWidget.js

```diff
@@ -0,0 +1,120 @@
+/*globals define, $, WebGMEGlobal*/
+define([
+    'js/Controls/PropertyGrid/Widgets/WidgetBase',
+    'js/logger'
+], function (
+    WidgetBase,
+    Logger
+) {
+    'use strict';
+
+    const BTN_ATTACH = $('<a class="btn btn-mini btn-dialog-open"><i class="glyphicon glyphicon-file"/></a>'),
+        INPUT_FILE_UPLOAD = $('<input type="file" />'),
+        ASSET_WIDGET_BASE = $('<div class="asset-widget" />'),
+        ASSET_LINK = $('<a class="local-download-link" href="" target="_blank"></a>');
+
+    class BrowserAssetWidget extends WidgetBase {
```

Comment:
Since this always runs in the browser, it might be better to name it something like `FileWidget` since it appears to just be a widget that provides a file input (at least that is the core functionality).

Reply:
Okay. This should have been addressed from the initial review. By browser `BrowserAssetWidget`, the intention was to make it clear that it is like the `AssetWidget` from upstream but rather that asset should be handled in the browser rather than via `BlobClient`.

Reply:
ah, that makes more sense. I still have a slight preference for something like `FileWidget` since I think it is a little clearer but I think we can leave it if you want.